YurisCodingClub/html-tag-quiz
# Update list of html tags

## Description
The index.js file contains the `tags` array holding a list of html tags. Originally, this list was incomplete (there were only 112 tags) and some tags had spelling errors.

As suggested in the issue, I used the W3 as my source for the complete list of html tags. I also used a list from MDN as a control list. Ensuring accuracy is crucial for this task, so I arrived at a new, complete list using a combination of manual and programmatic steps:
- copied and pasted the W3 and MDN lists into an Excel spreadsheet
- used a JavaScript formula to turn the lists into arrays (`CHAR(34) & Excell cell address for tag element & CHAR(34) & ","`)
- created a `tags.js` file to compare the W3 list against the original list, and also against the MDN list. I considered this step necessary to better understand the status of the original list and to make sure there were no errors from the first step.
- the current `tags` array in index.js is based on the W3 list with the `<>` removed. It contains 124 tags.
- refactored the code for checking if input value is in the `tags` array since there is no need to account for `<>`

**Note**
The MDN list contains 4 tags (`hgroup`, `menu`, `search`, `slot`) that are not labeled as deprecated by MDN, but they are not listed by W3. Please see the `tags.js` file for more details.

## Related Issue
Closes #9 

## Acceptance Criteria (AC)

- [x] All tags are written correctly
- [x] All tag strings only consist of words, no open close arrows
- [x] All html tags are in the list, current and outdated

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓  | :bug: Bug fix              |
|  ✓   | :hammer: Refactoring       |

## Screenshots
BEFORE the bug fix, the list would accept misspelled tags that were part of the array, but would not add tags like `title`, a valid tag originally not part of the list.
<img alt="list showing the following items: inputs, capton, articel" src="https://github.com/YurisCodingClub/html-tag-quiz/assets/115652409/963eddb7-d267-4c82-abf1-ac4d8a832b08">

AFTER
<img alt="list showing the following items: inputs, title" src="https://github.com/YurisCodingClub/html-tag-quiz/assets/115652409/eedf8676-418a-44a8-a85e-6b7a86f411d6">

## Testing Steps / QA Criteria
Example of tags in the original list that were misspelled: 'articel', `articele', 'capton'.
Example of valid tags not included in the original list: 'dir',  'font', 'frame'.

**To test:** 
- Type either one of the misspelled tags in the input field; clicking Submit should NOT add them to the list
- Type any of the newly added tags; clicking Submit should add them to the list
